crud/operations.go

```diff
@@ -23,11 +27,44 @@ const (
 
 // Operation describes CRUD operation as a table
 // {operator, field_identifier, value}.
+// Splice operation described as a table
+// {operator, field_identifier, position, length, replace_string}.
 type Operation struct {
-	// Instruct msgpack to pack this struct as array, so no custom packer
-	// is needed.
-	_msgpack struct{} `msgpack:",asArray"` //nolint: structcheck,unused
 	Operator Operator
 	Field    interface{} // Number or string.
 	Value    interface{}
+	// Pos, Len, Replace fields used in the Splice operation.
+	Pos     int
+	Len     int
+	Replace string
+}
+
+// EncodeMsgpack encodes Operation.
+func (o Operation) EncodeMsgpack(enc *msgpack.Encoder) error {
```

Comment:
Let's test it with a unit-test too.

Reply:
Added a new test to the `crud/operations_test.go`.